plugins/modules/templates_import.py

```diff
@@ -156,6 +173,8 @@ def main():
             force=dict(type='bool'),
             lock=dict(type='bool'),
             negate=dict(type='bool'),
+            http_proxy_policy=dict(choices=['global', 'none', 'selected']),
+            http_proxy=dict(type='entity'),
         ),
         supports_check_mode=False,
         required_plugins=[('templates', ['*'])],
```

Comment:
You can add
```suggestion
        required_plugins=[('templates', ['*'])],
        required_if=[('http_proxy_policy', 'selected', ('http_proxy'))],
```
to enforce the requirement at ansible level too